jwt/jwt.go

```diff
@@ -115,8 +121,14 @@ func ParseEncrypted(s string) (*JSONWebToken, error) {
 }
 
 // ParseSignedAndEncrypted parses signed-then-encrypted token from JWE form.
-func ParseSignedAndEncrypted(s string) (*NestedJSONWebToken, error) {
-	enc, err := jose.ParseEncrypted(s)
+//
+// The encryptionKeyAlgorithms and contentEncryption parameters are used to validate the "alg" and "enc"
+// header parameters, respectively, of the outer JWE. They must be nonempty, and each "alg" or "enc"
+// header in parsed data must contain a value that is present in the corresponding parameter. That
+// includes the protected and unprotected headers as well as all recipients. To accept
+// multiple algorithms, pass a slice of all the algorithms you want to accept.
+func ParseSignedAndEncrypted(s string, encryptionKeyAlgorithms []jose.KeyAlgorithm, contentEncryption []jose.ContentEncryption) (*NestedJSONWebToken, error) {
```

Comment:
I _suspect_ that the second parameter here is named `encryptionKeyAlgorithms` to differentiate it from a hypothetical `signingKeyAlgorithms`. Is that correct?

In that case, do you have any objection to giving this same name to the second parameter of `ParseEncrypted`? Even though that function doesn't have any hypothetical signing keys, I think that consistency of naming would be nice.

Reply:
Yep that's correct. And good idea.